addons/packages/vsphere-csi/2.4.1/bundle/config/vsphereconf.lib.txt

```diff
@@ -0,0 +1,42 @@
+
+(@ def vsphere_conf(values): -@)
+[Global]
+insecure-flag = true
+cluster-id = (@=values.vsphereCSI.namespace @)/(@=values.vsphereCSI.clusterName @)
+
+[VirtualCenter "(@=values.vsphereCSI.server @)"]
+user = "(@=values.vsphereCSI.username.replace("\\", "\\\\").replace("\"", "\\\"").replace("\t","\\t") @)"
+password = "(@=values.vsphereCSI.password.replace("\\", "\\\\").replace("\"", "\\\"").replace("\t","\\t") @)"
+datacenters = "(@=values.vsphereCSI.datacenter @)"
+
+[Network]
+public-network = "(@=values.vsphereCSI.publicNetwork @)"
+(@ if values.vsphereCSI.region or values.vsphereCSI.zone: -@)
+
+[Labels]
+(@ # if useTopologyFeature has defined and true is passed in, csi config uses topology-categories. Otherwise still use region,zone -@)
+(@ if hasattr(values.vsphereCSI, "useTopologyFeature") and values.vsphereCSI.useTopologyFeature: -@)
```

Comment:
Logic change in here
If `useTopologyFeature` is true, use `topology-categories`
else, use `region` and `zone`